client/src/components/Register.js

```diff
@@ -134,27 +136,67 @@ class Register extends Component {
       body: JSON.stringify(registrationData),
     })
       .then(response => {
-        response.json();
+        switch (response.status) {
+          case 201:
+            // Clear out the register props now - any process that returns the user to the
+            // register screen will need a new "start" call
+            this.setState({
+              registerProps: null,
+            });
+
+            this.setupBackupMethod();
+            return null;
+          default:
+        }
+        return response.json();
       })
-      .then(() => {
-        // If there's a backup method that's not registered then we initialise that
-        if (
-          this.shouldSetupBackupMethod()
-          && selectedMethod.urlSegment !== backupMethod.urlSegment
-        ) {
+      .then(response => {
+        // Failure states are captured here
+
+        // Expected structure: ['Registration failed', 'Because you misspelled your password.']
```

Comment:
Maybe we can make the server side have a more declarative structure? I'm not sure I like "assume the second item in the array is the thing we want".

Reply:
Yeah, the other places where `errors` is returned from the backend is pushing single string messages into its array. This particular case is returning two strings in the array, so it's different to the usual response format. I'll change it to match the rest.

Reply:
I've made it just return the error message from the authenticator, not the "Registration failure" message as well